torch/csrc/lazy/core/lazy_graph_executor.cpp

```diff
@@ -330,13 +330,20 @@ bool LazyGraphExecutor::DataCacheArena::TensorComparer::operator()(
 
 auto LazyGraphExecutor::DataCacheArena::GetDataCache(
     const BackendDevice& device) -> DataCache* {
+  
   std::lock_guard<std::mutex> lock(mutex_);
-  auto it = device_caches_.find(device);
-  if (it == device_caches_.end()) {
-    std::unique_ptr<DataCache> cache(new DataCache(max_cache_size_));
-    it = device_caches_.emplace(device, std::move(cache)).first;
+  if (FLAGS_torch_lazy_enable_device_data_cache) {
+    auto it = device_caches_.find(device);
+    if (it == device_caches_.end()) {
+      std::unique_ptr<DataCache> cache(new DataCache(max_cache_size_));
+      it = device_caches_.emplace(device, std::move(cache)).first;
+    }
+    return it->second.get();
+  } else {
+      // If cache is disabled then always return a zero size cache
```

Comment:
should this invariant be protected (that nobody externally adds to this supposedly empty cache)?

Reply:
I think I am missing something.  Creation is protected under the class mutex.  For all devices in this configuration a single zero sized cache is returned (irrespective of backend device since caches are disabled / hidden).

The change assumes that GetDataCache is always used as the point of entry.  The other data caches are intentionally left alone so that we can re-enable them.

Reply:
oh, i meant in case some code outside called this, got the zero sized cache, then resized it or something. but i suppose that's a bit adversarial and far fetched